src/lib/Timeline.js

```diff
@@ -340,14 +335,39 @@ export default class ReactCalendarTimeline extends Component {
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    const { visibleTimeStart, visibleTimeEnd, items, groups } = nextProps
+    const { visibleTimeStart, visibleTimeEnd, items, groups, zoom } = nextProps
 
     // This is a gross hack pushing items and groups in to state only to allow
     // For the forceUpdate check
     let derivedState = { items, groups }
 
     // if the items or groups have changed we must re-render
-    const forceUpdate = items !== prevState.items || groups !== prevState.groups
+    const forceUpdate =
+      items !== prevState.items ||
+      groups !== prevState.groups ||
+      zoom !== oldZoom
+
+    const oldZoom = prevState.visibleTimeEnd - prevState.visibleTimeStart
+
+    if (zoom !== oldZoom) {
+      const zoomDiff = oldZoom - zoom
+      const newVisibleTimeStart = Math.round(
+        prevState.visibleTimeStart + zoomDiff * 0.5
+      )
+
+      Object.assign(
+        derivedState,
+        calculateScrollCanvas(
+          newVisibleTimeStart,
+          newVisibleTimeStart + zoom,
```

Comment:
Could you put this in a variable that indicates that this is the new end? Would make it a lot easier to understand